### PR TITLE
Drop support for Python versions earlier than 3.8

### PR DESCRIPTION
**Pull request by mdickinson**
This PR drops support for Python versions older than 3.8.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -276,13 +276,6 @@ def get_long_description():
         Operating System :: POSIX :: Linux
         Programming Language :: Python
         Programming Language :: Python :: 3
-        Programming Language :: Python :: 3.6
```
Comment (mdickinson) [inline setup.py:279]: Removing these classifiers - they're a pain to maintain and provide little value. Note that we do have a `python_requires` keyword below giving the minimum supported version of Python.

File: setup.py [hunk 2]
```diff
@@ -317,11 +305,9 @@ def get_long_description():
             "flake8",
             "flake8-ets",
             "mypy",
-            # NumPy is not yet available for Python 3.12, but that should be
-            # fixed soon: https://github.com/numpy/numpy/issues/23808
-            "numpy; python_version < '3.12'",
+            "numpy",
             "pyface",
-            "PySide6; python_version >= '3.7' and python_version < '3.12'",
+            "PySide6; python_version < '3.12'",
```
Comment (mdickinson) [inline setup.py:310]: It's _possible_ that we don't need this restriction any more.

**Comment by mdickinson** [comment]
Hmm. A slightly odd failure on Python 3.13 / Windows:
```
LINK : fatal error LNK1104: cannot open file 'python313t.lib'
```

**Comment by mdickinson** [comment]
> A slightly odd failure on Python 3.13 / Windows:

Looks like this might be the cause: https://github.com/python/cpython/issues/115582

File: setup.py (hunk 2 above)
Comment (corranwebster) [reply setup.py]: The most recent release (6.6.2) claims compatibility with Python 3.12.  I wouldn't expect it to have support for 3.13.

Comment (mdickinson) [reply setup.py]: Yep, the tests seem fine with Python 3.12. I updated the restriction in [7c1306f](https://github.com/enthought/traits/pull/1773/commits/7c1306f181dbaa38ddc6e1d17631e212af147008)

**Comment by mdickinson** [comment]
I'm not sure what to do about the Windows failure. It should be fixed as soon as Python 3.13.0a5 is released and is available for use by the `setup-python` GitHub Action. 3.13.0a5 is expected in around a week from now (according to [PEP 719](https://peps.python.org/pep-0719/)).

File: .github/workflows/run-core-traits-tests.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
```
Comment (mdickinson) [inline .github/workflows/run-core-traits-tests.yml:14]: Pinning to the latest working version of 3.13.0.

```suggestion
        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13.0-alpha.3']
```

**Comment by mdickinson** [comment]
@corranwebster I think this is ready for review. I've pinned the Python 3.13 version being used for the core tests; we'll need to remove that pin at some point.

I'll need to adjust the PR expected checks to remove the 3.7 versions; I'll do that just before merging.

**Review by corranwebster — APPROVED** [review]
Looks good.

**Pull request merged**